plugins/aws/CodeDeploy.java

```
@@ -0,0 +1,108 @@
+                    " for some reason.");
+            System.out.println("Error Message:    " + ase.getMessage());
+            System.out.println("HTTP Status Code: " + ase.getStatusCode());
+            System.out.println("AWS Error Code:   " + ase.getErrorCode());
+            System.out.println("Error Type:       " + ase.getErrorType());
+            System.out.println("Request ID:       " + ase.getRequestId());
+        } catch (AmazonClientException ace) {
+            System.out.println("Caught an AmazonClientException, which " +
+                    "means the client encountered " +
+                    "an internal error while trying to " +
+                    "communicate with S3, " +
+                    "such as not being able to access the network.");
+            System.out.println("Error Message: " + ace.getMessage());
+        }
+    }
+
+    protected static void zip(String fileName, ArrayList<String> files) {
+            byte[] buffer = new byte[1024];            
+            try {
+                FileOutputStream fos = new FileOutputStream("/tmp/" + fileName + ".zip");
+                ZipOutputStream zos = new ZipOutputStream(fos);
+                for (String file: files) {
+                    try {
+                        File fs = new File(file);
+                        System.out.println(fs.getName());
+                        FileInputStream fis = new FileInputStream(fs);
+                        ZipEntry zipEntry = new ZipEntry(fs.getName());
+                        zos.putNextEntry(zipEntry);
+
+                        byte[] bytes = new byte[1024];
+                        int length;
+                        while ((length = fis.read(bytes)) >= 0) {
+                            zos.write(bytes, 0, length);
+                        }
+
+                        zos.closeEntry();
+                        fis.close();
+                    } catch (Exception e) {
+
+                    }
```

catching exceptions and ignoring them is almost always a bad idea in my opinion. hides errors.